### PR TITLE
[V2] chore: Create helm charts and release files for v2.0.0-beta.1 release

### PR DESCRIPTION
**Pull request by nearora-msft**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: charts/v2.0.0-beta.1/csi-azuredisk-controller.yaml [hunk 1]
```diff
@@ -0,0 +1,181 @@
+---
```
Comment (edreed) [inline charts/v2.0.0-beta.1/csi-azuredisk-controller.yaml:1]: There should be any yaml files directly under `charts/v2.0.0-beta.1`

**Comment by edreed** [comment]
The [`deploy/example/failover/README.md`](https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/main_v2/deploy/example/failover/README.md#demo) also needs to be updated with the new release version.

**Comment by nearora-msft** [comment]
> The [`deploy/example/failover/README.md`](https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/main_v2/deploy/example/failover/README.md#demo) also needs to be updated with the new release version.

Updated

**Pull request merged**